tensorrt_llm/serve/openai_server.py

```
@@ -2182,4 +2180,13 @@ async def __call__(self,
                                 log_level="info",
                                 timeout_keep_alive=TIMEOUT_KEEP_ALIVE)
-        await uvicorn.Server(config).serve(sockets=sockets)
+        server = uvicorn.Server(config)
+
+        async def _register_after_serving():
+            while not server.started:
+                await asyncio.sleep(0.1)
+            if self.disagg_cluster_worker:
+                await self.disagg_cluster_worker.register_worker()
+
+        asyncio.create_task(_register_after_serving())
```

Ruff (0.15.13)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)